components/posts/posts.php

```diff
@@ -253,6 +274,277 @@ function mm_posts_output_masonry_sizers() {
  * @param  object  $context  The global post object.
  * @param  array   $args     The instance args.
  */
+function mm_post_article( $post, $context, $args ) {
+
+	ob_start(); ?>
+
+	<article id="post-<?php the_ID( $post ); ?>" <?php post_class( 'mm-post' ); ?> itemscope itemtype="http://schema.org/BlogPosting" itemprop="blogPost" aria-label="Article">
+
+		<?php do_action( 'mm_posts_header', $post, $context, $args ); ?>
+
+		<?php do_action( 'mm_posts_content', $post, $context, $args ); ?>
+
+		<?php do_action( 'mm_posts_footer', $post, $context, $args ); ?>
+
+	</article>
+
+	<?php return ob_get_clean();
+}
+
+/**
+ * Output a taxonomy term filter.
+ *
+ * @since   1.0.0
+ *
+ * @param   WP_Query  $query    The query object.
+ * @param   WP_Post   $context  The global post object.
+ * @param   array     $args     The instance args.
+ *
+ * @return  string              The category filters HTML.
+ */
+function mm_posts_output_taxonomy_term_filter( $query, $context, $args ) {
+
+	$taxonomy = $args['taxonomy'] ? ( $args['taxonomy'] ) : 'category';
+
+	ob_start(); ?>
+
+	<?php if ( $args['filter_style'] == 'dropdown' ) : ?>
+
+		<div class="mm-posts-filter-wrapper">
```

Comment:
I'd put this wrapping markup outside of the if statement here. Ideally the if statement would only wrap `.mm-posts-filter`. Also, try to maintain the hierarchy with indentation if you can, i.e.:

```
<div class="mm-posts-filter-wrapper">
    <div class="mm-posts-filter">
        <?php // Do stuff... ?>
    </div>
</div>
```